### ISSUE TITLE
getDeviceState not working on Android 

### ISSUE THREAD

**Issue by GeceGibi** [OneSignal/OneSignal-Flutter-SDK#466]
**Description:**

Other functions works fine but `OneSignal.shared.getDeviceState()` not working.

**Environment**

OneSignal Version: 3.2.1

Device: `Redmi Note 9S (mobile) • 2f755042 • android-arm64  • Android 11 (API 30)`

I'm trying now with other devices if works with any device i will update this issue.

**Comment by GeceGibi** [comment]
Error log: `type 'Null' is not a subtype of type 'bool' in type cast`

**Comment by GeceGibi** [comment]
This line is broken

 ```dart
json['isPushDisabled'] as bool
``` need to be  ```dart
json['pushDisabled'] as bool
```

https://github.com/OneSignal/OneSignal-Flutter-SDK/blob/51d585092b3a9dce26553ec4110affab4761565f/lib/src/permission.dart#L103

**Comment by GeceGibi** [comment]
and this line 
need to be  ```json['`subscribed'] as bool```

https://github.com/OneSignal/OneSignal-Flutter-SDK/blob/51d585092b3a9dce26553ec4110affab4761565f/lib/src/permission.dart#L104

**Comment by charlesarchibong** [comment]
I am facing the same issue here

**Comment by GeceGibi** [comment]
Issue fixed with new release [3.2.2](https://github.com/OneSignal/OneSignal-Flutter-SDK/releases/tag/3.2.2)
